src/main/java/com/ita/if103java/ims/dao/impl/WarehouseDaoImpl.java

```diff
@@ -0,0 +1,167 @@
+package com.ita.if103java.ims.dao.impl;
+
+import com.ita.if103java.ims.dao.WarehouseDao;
+import com.ita.if103java.ims.entity.Warehouse;
+import com.ita.if103java.ims.exception.CRUDException;
+import com.ita.if103java.ims.exception.EntityNotFoundException;
+import com.ita.if103java.ims.mapper.jdbc.WarehouseRowMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class WarehouseDaoImpl implements WarehouseDao {
+    private static final Logger LOGGER = LoggerFactory.getLogger(WarehouseDaoImpl.class);
+    private JdbcTemplate jdbcTemplate;
+    private WarehouseRowMapper warehouseRowMapper;
+
+
+    @Autowired
+    public WarehouseDaoImpl(DataSource dataSource, WarehouseRowMapper warehouseRowMapper) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+        this.warehouseRowMapper = warehouseRowMapper;
+    }
+
+    private PreparedStatement createWarehouseStatement(Warehouse warehouse, Connection connection) throws SQLException {
+        int i = 0;
+        PreparedStatement statement = connection.prepareStatement(QueryToWarehouse.SQL_CREATE_USER,
+            PreparedStatement.RETURN_GENERATED_KEYS);
+        statement.setString(++i, warehouse.getName());
+
+        statement.setString(++i, warehouse.getInfo());
+
+        statement.setInt(++i, warehouse.getCapacity());
+
+        statement.setBoolean(++i, warehouse.isBottom());
+
+        statement.setLong(++i, warehouse.getParentID());
+
+        statement.setLong(++i, warehouse.getAccountID());
+
+        statement.setLong(++i, warehouse.getTopWarehouseID());
+
+        statement.setBoolean(++i, warehouse.isActive());
+
```

Comment:
please format your code, Ctrl+Alt+L
Do not add blank line after each row, avoid double blank lines, add blank line after class declaration